sqlx-postgres/src/connection/stream.rs

```diff
@@ -53,13 +53,14 @@ macro_rules! read_packet {
         loop {
             read_packet!($(@$blocking)? @stream $self, 0, 5);
 
-            let mut header: Bytes = $self.stream.take(5);
+            // peek at the messaage type and payload size
+            let r#type = MessageType::try_from(*$self.stream.get(0,1))?;
+            let size = (u32::from_le_bytes($self.stream.get(1,4)) - 4) as usize;
```

Comment:
Postgres uses big endian.

Reply:
I should have caught this.  Misread the other instances.  Thanks!